### PR TITLE
feat: Support "cancelled" and "failed" Shutdown operations [MLG-468]

### PR DESCRIPTION
**Pull request by tayritenour**
## Description

[MLG-468]

Supports sending Shutdown operations that are "cancelled" or "failed". This allows us on the python harness side to send operations like `searcher.Shutdown(failure=True)` inside of a Custom Searcher `SearchMethod` to signal to the master that an experiment should be considered failed.

Before, we made the assumption that any failures by individual trials in Custom Searches were all equally valid. However in use cases like the Deepspeed Autotuning project, we have special trials that cannot fail or the whole job should be stopped and considered failed. This change would allow the user to control that.

## Test Plan

Here is a minimal example of a Custom Searcher that can simulate the new functionality:
[test_shutdown.zip](https://github.com/determined-ai/determined/files/11315591/test_shutdown.zip)

To run, try both:
`det experiment create dummy_searcher/no_fail.yaml .`
and
`det experiment create dummy_searcher/fail.yaml .`

Screenshot 2023-04-24 at 5 52 36 PM

Each Custom Searcher should create two experiments, one being the orchestrator of the other agent experiment, which actually runs the prospective trials. The orchestrator will schedule a single trial, and on close, choose to `Shutdown` the agent experiment with a failure state or not depending on the yaml. Only the agent experiment should be notated as "failed".

## Checklist

- [ ] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

## Ticket

[MLG-468]: https://hpe-aiatscale.atlassian.net/browse/MLG-468?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: proto/src/determined/experiment/v1/searcher.proto [hunk 1]
```diff
@@ -165,7 +165,12 @@ message CloseTrialOperation {
 // Shut down custom searcher method.
 message ShutDownOperation {
   // Cannot have an empty message type.
+  // Deprecated: But maintained here for backwards compatibility
   int32 placeholder = 1;
```
Comment (rb-determined-ai) [inline proto/src/determined/experiment/v1/searcher.proto:169]: It's an internal API, so I vote to just break it.  Also the comment above this line.

**Review by stoksc — APPROVED** [review]
lgtm

File: proto/src/determined/experiment/v1/searcher.proto (hunk 1 above)
Comment (stoksc) [reply proto/src/determined/experiment/v1/searcher.proto:169]: i would agree to remove this.

Comment (tayritenour) [reply proto/src/determined/experiment/v1/searcher.proto]: Makes sense, testing the changes to remove this

**Pull request merged**